dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/internal/OkHttpUploaderTest.kt

```diff
@@ -275,6 +279,105 @@ internal class OkHttpUploaderTest {
             )
     }
 
+    @Test
+    fun `M throw a MaxSizeExceededException W upload() { mappingFile size exceeded 50 MB}`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val fakeTooLargeMappingFile = spy(File(tempDir, forge.anAlphabeticalString()))
+        doReturn(1024L * 1024L * 50 + forge.anInt(min = 0))
+            .whenever(fakeTooLargeMappingFile).length()
+
+        // THEN
+        assertThrows<MaxSizeExceededException>(
+            {
+                testedUploader.upload(
+                    mockWebServer.url("/").toString(),
+                    fakeTooLargeMappingFile,
+                    fakeRepositoryFile,
+                    fakeIdentifier,
+                    fakeRepositoryInfo
+                )
+            }
+        )
+    }
+
+    @Test
+    fun `M throw a MaxSizeExceededException W upload() { mappingFile size between 50 and 51 MB}`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val fakeTooLargeMappingFile = spy(File(tempDir, forge.anAlphabeticalString()))
+        val megabyte = 1024L * 1024L
+        val size = megabyte * 50 + forge.anInt(min = 1, max = megabyte.toInt())
+        doReturn(size)
+            .whenever(fakeTooLargeMappingFile).length()
+
+        // THEN
+        assertThrows<MaxSizeExceededException>(
+            {
+                testedUploader.upload(
+                    mockWebServer.url("/").toString(),
+                    fakeTooLargeMappingFile,
+                    fakeRepositoryFile,
+                    fakeIdentifier,
+                    fakeRepositoryInfo
+                )
+            }
+        )
+    }
+
+    @Test
+    fun `M fail and do nothing W upload() { mappingFile size exceeded 50 MB }`(
+        forge: Forge
+    ) {
+        // GIVEN
+        val fakeTooLargeMappingFile = spy(File(tempDir, forge.anAlphabeticalString()))
+        doReturn(forge.aLong(min = 51))
+            .whenever(fakeTooLargeMappingFile).length()
```

Comment:
Aren't you faking a file with size ≥ 51 bytes? How can you be sure that it exceeds 50Mb ? Also isn't this redundant with the test above?